lib/vector-core/src/enrichment_table.rs

```diff
@@ -0,0 +1,5 @@
+use std::collections::BTreeMap;
+
+pub trait EnrichmentTable {
```

Comment:
Do we want some kind of query-hint method that will get called during remap compilation? That seems like it could be a non-trivial portion of the implementation, so it seem worth exploring early.